ansible/roles/test/files/ptftests/radv_ipv6_ra_test.py

```diff
@@ -0,0 +1,240 @@
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "P")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "res")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "routerlifetime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "reachabletime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6ND_RA, "retranstimer")
+
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "prefixlen")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "L")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "A")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "R")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "res1")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "validlifetime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo,
+                                    "preferredlifetime")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "res2")
+        masked_rapkt.set_do_not_care_scapy(scapy.layers.inet6.ICMPv6NDOptPrefixInfo, "prefix")
+
+        return masked_rapkt
+
+"""
+@summary: This test is to validate the unsolicted router advertisements sent on
+the VLAN network of the ToR. In this test we listen on the first PTF port
+(Eg eth0) for any RA messages.
+
+"""
+
+class RadvUnSolicitedRATest(DataplaneBaseTest):
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def setUp(self):
+        DataplaneBaseTest.setUp(self)
+
+        self.test_params = testutils.test_params_get()
+
+        self.hostname = self.test_params['hostname']
+        self.downlink_vlan_mac = self.test_params['downlink_vlan_mac']
+        self.downlink_vlan_ip6 = self.test_params['downlink_vlan_ip6']
+        self.ptf_port_index = int(self.test_params['ptf_port_index'])
+        self.ptf_port_mac = self.dataplane.get_mac(0, self.ptf_port_index)
+        self.radv_max_ra_interval = int(self.test_params['max_ra_interval'])
+
+        rapkt = self.create_icmpv6_router_advertisement_packet(
+                                                src_mac=self.downlink_vlan_mac,
+                                                dst_mac=ALL_NODES_MULTICAST_MAC_ADDRESS,
+                                                src_ip=self.downlink_vlan_ip6,
+                                                dst_ip=ALL_NODES_IPV6_MULTICAST_ADDRESS)
+        self.masked_rapkt = self.mask_off_dont_care_ra_packet_fields(rapkt)
+
+
+    def tearDown(self):
+        DataplaneBaseTest.tearDown(self)
+
+    """
+    @summary: Verify received RA packet
+
+    """
+
+    def verify_periodic_router_advertisement(self):
+        testutils.verify_packet(
```

Comment:
Suggest placing first argument on this line (e.g., `testutils.verify_packet(self,`